pkg/service/model_build_load_balancer.go

```diff
@@ -267,6 +271,16 @@ func (t *defaultModelBuildTask) buildLoadBalancerSubnets(ctx context.Context, sc
 		)
 	}
 
+	// for internet-facing Load Balancers, the subnets mush have at least 8 available IP addresses;
+	// for internal Load Balancers, this is only required if private ip address is not assigned
```

Comment:
why not check if the service has the  `service.beta.kubernetes.io/aws-load-balancer-private-ipv4-addresses` annotation set in this function? If private annotation is set on an internal load balancer, we can skip the min available check. If we check for the annotation here, then this function doesn't need the ipv4Configured arg to be passed in. This will also help us decouple the subnet discovery logic from the build load balancer function - and possibly less change :).

Reply:
changes committed, thanks